generic3g/InfoUtilities.F90

```diff
@@ -0,0 +1,275 @@
+#include "MAPL_Generic.h"
+
+! This module is intended to manage user-level access to ESMF info
+! objects and thereby ensure consistent support for namespace
+! management and such.
+
+module mapl3g_InfoUtilities
+   use mapl_ErrorHandling
+   use mapl_KeywordEnforcer
+
+   use mapl3g_esmf_info_keys
+   use esmf, only: ESMF_StateItem_Flag
+   use esmf, only: ESMF_STATEITEM_FIELD
+   use esmf, only: operator(==), operator(/=)
+   use esmf, only: ESMF_Info
+   use esmf, only: ESMF_InfoGetFromHost
+   use esmf, only: ESMF_InfoGet
+   use esmf, only: ESMF_InfoGetCharAlloc
+   use esmf, only: ESMF_InfoSet
+   use esmf, only: ESMF_State
+   use esmf, only: ESMF_StateGet
+   use esmf, only: ESMF_Field
+   use esmf, only: ESMF_KIND_I4
+
+   implicit none
+   private
+
+   public :: MAPL_InfoGetShared
+   public :: MAPL_InfoSetShared
+   public :: MAPL_InfoGetPrivate
+   public :: MAPL_InfoSetPrivate
+
+   interface MAPL_InfoGetShared
+      procedure :: info_get_shared_string
+      procedure :: info_get_shared_i4
+      procedure :: info_get_state_shared_string
+      procedure :: info_get_stateitem_shared_i4
+   end interface MAPL_InfoGetShared
+
+   interface MAPL_InfoSetShared
+      procedure :: info_set_shared_string
+      procedure :: info_set_shared_i4
+      procedure :: info_set_state_shared_string
+      procedure :: info_set_stateitem_shared_i4
+   end interface MAPL_InfoSetShared
+   
+   interface MAPL_InfoGetPrivate
+      procedure :: info_get_private_i4
+      procedure :: info_get_stateitem_private_i4
+   end interface MAPL_InfoGetPrivate
+   
+   interface MAPL_InfoSetPrivate
+      procedure :: info_set_private_i4
+       procedure :: info_set_stateitem_private_i4
+   end interface MAPL_InfoSetPrivate
+
+
+contains
+
+   subroutine info_get_shared_string(info, key, value, unusable, rc)
```

Comment:
Suggestion: Use the `ESMF_InfoIsPresent` function to distinguish between a missing key and other ESMF error.